src/timepicker/TimePickerState.ts

```diff
@@ -1,68 +1,125 @@
-/* eslint-disable @typescript-eslint/no-non-null-assertion */
-import React from "react";
-import { useCompositeState, useDisclosureState } from "reakit";
+import {
+  PopoverInitialState,
+  usePopoverState,
+  unstable_useId as useId,
+} from "reakit";
+import * as React from "react";
+import { InputBase } from "@react-types/shared";
+import { useControllableState } from "@chakra-ui/hooks";
+
+import { useSegmentState } from "../segment-spinner";
 import { useTimePickerColumnState } from "./TimePickerColumnState";
 
-export interface TimePickerStateProps {
-  onChange?: (v: Date | null) => void;
-  visible?: boolean;
+export interface TimePickerStateProps extends PopoverInitialState, InputBase {
   value?: string;
+  defaultValue?: string;
+  onChange?: (v: string) => void;
+  formatOptions?: Intl.DateTimeFormatOptions;
```

Comment:
Add the timeStyle / dateStyle bindings

Reply:
We need to add and make a type for ourselves.

I think we can better add `utils/type.ts` for common types among components like other libraries.

Reply:
Yeah